exercises/rail-fence-cipher/.meta/src/reference/java/RailFenceCipher.java

```diff
@@ -0,0 +1,74 @@
+import java.util.ArrayList;
+import java.util.List;
+
+class RailFenceCipher {
+
+    private int noOfRails;
+
+    RailFenceCipher(int noOfRails) {
+        this.noOfRails = noOfRails;
+    }
+
+    String getEncryptedData(String plainText) {
+        List<String> railFence = new ArrayList<>();
+        for (int i = 0; i < noOfRails; i++) {
+            railFence.add("");
+        }
+
+        int number = 0, increment = 1;
+        for (char c : plainText.toCharArray()) {
+            if (number + increment == noOfRails) {
+                increment = -1;
+            } else if (number + increment == -1) {
+                increment = 1;
+            }
+            String temp = railFence.get(number);
+            railFence.remove(number);
+            railFence.add(number, temp + c);
+            number += increment;
+        }
+        String buffer = "";
+        for (String s : railFence) {
+            buffer = buffer.concat(s);
+        }
+        return buffer;
+    }
+
+    String getDecryptedData(String cipherText) {
+        String decrypted;
+        ArrayList<ArrayList<Integer>> railFence = new ArrayList<>(noOfRails);
```

Comment:
Is there any particular reason you're storing each rail as a list of integers? You could just store it as a string and append characters to it as you go through the ciphertext. That should make it easier to get the final output at the end :)

Reply:
I've stored it in a kind of matrix form for a 2 loop iteration

Reply:
Sure, but since you're dealing with characters it makes more sense to store it as characters rather than integers. You can have a list of strings and loop through the characters of each string.

And as a tip for the future, it would be really helpful if you could respond to reviews relatively soon after we do them. For big reviews like this it takes time for me to get into the mindset of the code and understand what you're doing. If you had made these comments a month ago when I first reviewed this code I would have been able to give way more useful answers. And I'm guessing it would be easier for you too to change your code right after you wrote it instead of more than a month after.
I know you're probably very busy with work and things, but in that case it might be easier for you to work on smaller issues rather than trying to implement a whole exercise? Like I said, this is just a suggestion for the future since PRs that remain open for months become really hard to review. But I do really appreciate you trying to help improve the track :)